distsql/distsql.go

```diff
@@ -141,46 +144,57 @@ func (r *selectResult) NextRaw() ([]byte, error) {
 }
 
 // NextChunk reads data to the chunk.
-func (r *selectResult) NextChunk(goCtx goctx.Context, chk *chunk.Chunk) error {
+func (r *selectResult) NextChunk(goCtx goctx.Context, chk *chunk.Chunk) (int64, error) {
 	chk.Reset()
+	count := int64(0)
 	for chk.NumRows() < r.ctx.GetSessionVars().MaxChunkSize {
 		if r.selectResp == nil || r.respChkIdx == len(r.selectResp.Chunks) {
-			err := r.getSelectResp()
-			if err != nil || r.selectResp == nil {
-				return errors.Trace(err)
+			cnt, err := r.getSelectResp()
+			if err != nil {
+				return -1, errors.Trace(err)
+			}
+			count += cnt
+			if r.selectResp == nil {
+				return count, nil
 			}
 		}
 		err := r.readRowsData(chk)
 		if err != nil {
-			return errors.Trace(err)
+			return -1, errors.Trace(err)
 		}
 		if len(r.selectResp.Chunks[r.respChkIdx].RowsData) == 0 {
 			r.respChkIdx++
 		}
 	}
-	return nil
+	return count, nil
 }
 
-func (r *selectResult) getSelectResp() error {
+func (r *selectResult) getSelectResp() (int64, error) {
 	r.respChkIdx = 0
+	count := int64(0)
 	for {
 		re := <-r.results
 		if re.err != nil {
-			return errors.Trace(re.err)
+			return -1, errors.Trace(re.err)
 		}
 		if re.result == nil {
 			r.selectResp = nil
-			return nil
+			return count, nil
 		}
 		r.selectResp = new(tipb.SelectResponse)
 		err := r.selectResp.Unmarshal(re.result)
 		if err != nil {
-			return errors.Trace(err)
+			return -1, errors.Trace(err)
+		}
+		if len(r.selectResp.OutputCounts) > 0 {
+			count += r.selectResp.OutputCounts[0]
+		} else {
+			count = -1
```

Comment:
Why not just return an error?

Reply:
If the tikv is old, we may not have the `OutputCounts` set.